File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int answer=0;
        int buy = prices[0];
        for( int i=1 ;i <n;i++){
            if( prices[i] < buy) {
                buy = prices[i];
            }
            int diff = prices[i] - buy;
            answer = max(answer,diff);
        }
        return answer;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit=0
        diff=0
        buy = prices[0]
        for num in prices[1:]:
            if num < buy:
                buy = num
            else:
                diff = num - buy
            profit = max(profit,diff)
        return profit
